src/server/server/dal.py

```diff
@@ -287,15 +289,16 @@ def get_user_actions_on_docs(self, users, from_datetime):
         # this second point would be solved if the first one (filter on user range) is solved
 
         # 2) retrieve docs present in actions from database
-        doc_keys = list(set(action['document_key'] for action in db_actions))
-        docs = self._dal_doc._to_docs(doc_keys)  # pylint: disable=protected-access
-        doc_key_to_doc = dict(zip(doc_keys, docs))
+        doc_url_hashes = list(set(action['document_url_hash'] for action in db_actions))
```

Comment:
as for previously doc_keys, to fix iteration order for zip(doc_url_hashes, docs). It will work keeping the same set, but it might seem dubious to rely on set iteration order (which is not specified)
http://stackoverflow.com/questions/3848091/python-set-iteration-order-varies-from-run-to-run